### PR TITLE
[Merged by Bors] - Update operator-rs to 0.27.1

### PR DESCRIPTION
**Pull request by maltesander**
# Description

fixes: https://github.com/stackabletech/kafka-operator/issues/516

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/24/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator/src/kafka_controller.rs [hunk 1]
```diff
@@ -55,16 +55,15 @@ use std::{
...
+    command::{self, get_svc_container_cmd_args, kcat_container_cmd_args},
     discovery::{self, build_discovery_configmaps},
     pod_svc_controller,
-    utils::{self, ObjectRefExt},
+    utils::{self, build_recommended_labels, ObjectRefExt},
     ControllerConfig,
 };
 
-const RESOURCE_SCOPE: &str = "kafka-operator_kafkacluster";
+pub const KAFKA_CONTROLLER_NAME: &str = "kafkacluster";
```
Comment (razvan) [inline rust/operator/src/kafka_controller.rs:66]: Isn't the controller name `kafka-controller` ?

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ use std::collections::BTreeMap;
 use strum::{Display, EnumIter, EnumString};
 
 pub const APP_NAME: &str = "kafka";
+pub const OPERATOR_NAME: &str = "kafka.stackable.tech";
```
Comment (razvan) [inline rust/crd/src/lib.rs:32]: Isn't the operator name `kafka-operator` ?

File: rust/operator/src/pod_svc_controller.rs [hunk 3]
```diff
@@ -11,10 +11,10 @@ use stackable_operator::{
 use std::{sync::Arc, time::Duration};
 use strum::{EnumDiscriminants, IntoStaticStr};
 
+pub const POD_SERVICE_CONTROLLER_NAME: &str = "pod-service";
```
Comment (razvan) [inline rust/operator/src/pod_svc_controller.rs:14]: The other controller is called "kafkacluster" (which I also find confusing). Maybe rename both for consistency?

**Review by razvan — APPROVED** [review]
lgtm, but naming of constants is confusing.

File: rust/operator/src/kafka_controller.rs (hunk 1 above)
Comment (maltesander) [reply rust/operator/src/kafka_controller.rs:66]: Its the controller responsible for the kafkacluster. What is confusing about this?

File: rust/operator/src/pod_svc_controller.rs (hunk 3 above)
Comment (maltesander) [reply rust/operator/src/pod_svc_controller.rs:14]: This was just using the older `RESOURCE_SCOPE` naming. The controller responsible for the pod-service?

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (maltesander) [reply rust/crd/src/lib.rs:32]: What would you suggest? This is consistent for all the bumps and labels in other operators. Not to kin to change this everywhere?

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**